extensions/amp-ad-network-cloudflare-impl/0.1/amp-ad-network-cloudflare-impl.js

```diff
@@ -42,10 +43,23 @@ export class AmpAdNetworkCloudflareImpl extends AmpA4A {
    */
   isValidElement() {
     const el = this.element;
-    return this.isAmpAdElement()
-      && el.hasAttribute('src')
-      && el.hasAttribute('data-cf-network')
-      && NETWORKS[el.getAttribute('data-cf-network')] != null;
+    if (!(this.isAmpAdElement() && el.hasAttribute('src')
+        && el.hasAttribute('data-cf-network'))) {
+      return false;
+    }
+
+    const network = NETWORKS[el.getAttribute('data-cf-network')];
+    if (!network) {
+      return false;
+    }
+
+    const src = el.getAttribute('src');
+    if (!src.startsWith(network.base)) {
```

Comment:
please use `startsWith` from `string.js`. i guess the native one is not widely supported